ufl/algorithms/__init__.py

```diff
@@ -37,6 +37,8 @@
     "expand_derivatives",
     "extract_coefficients",
     "strip_variables",
+    "strip_form_arguments",
```

Comment:
Shouldn't it be named `strip_form_data` etc.?

Reply:
I called it this because ' form data' has a different meaning in UFL already (`ufl/algorithms/formdata.py`) so I didn't want a conflict. Perhaps I could call it `strip_form_argument_data`?

Reply:
FWIW, I don't like "arguments" since that suggests that coefficients are not stripped.

How about `detach_terminals` ?

Reply:
I don't think 'detach' is right because that doesn't make it clear that the terminals are getting replaced. `strip_terminal_data`?